core/src/main/java/hudson/cli/ConnectNodeCommand.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.cli;
+
+import hudson.Extension;
+import hudson.model.Computer;
+import hudson.security.AccessDeniedException2;
+import hudson.util.EditDistance;
+import jenkins.model.Jenkins;
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.Option;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * @author pjanouse
+ * @since TODO
+ */
+@Extension
+public class ConnectNodeCommand extends CLICommand {
+
+    @Argument(metaVar="NAME", usage="Slave name, or empty string for master", required=true, multiValued=true)
+    private List<String> nodes;
+
+    @Option(name="-f", usage="Cancel any currently pending connect operation and retry from scratch")
+    public boolean force = false;
+
+    private static final Logger LOGGER = Logger.getLogger(ConnectNodeCommand.class.getName());
+
+    @Override
+    public String getShortDescription() {
+
+        return Messages.ConnectNodeCommand_ShortDescription();
+    }
+
+    @Override
+    protected int run() throws Exception {
+
+        boolean errorOccurred = false;
+        final Jenkins jenkins = Jenkins.getInstance();
+
+        if (jenkins == null) {
+            stderr.println("The Jenkins instance has not been started, or was already shut down!");
+            return -1;
+        }
+
+        final HashSet<String> hs = new HashSet<String>();
+        hs.addAll(nodes);
+
+        List<String> names = null;
+
+        for (String node_s : hs) {
+            Computer item = null;
+
+            try {
+                item = jenkins.getComputer(node_s);
+
+                if(item == null) {
+                    if(names == null) {
+                        names = new ArrayList<String>();
+                        for (Computer c : jenkins.getComputers())
+                            if (c.getName().length() > 0)
```

Comment:
BTW do you know the JDK impl. of String.isEmpty() is just:
    public boolean isEmpty() {
        return this.value.length == 0;
    }

;-)